pkg/tfvars/libvirt/libvirt.go

```diff
@@ -34,21 +39,44 @@ func TFVars(masterConfig *v1beta1.LibvirtMachineProviderConfig, osImage string,
 		return nil, err
 	}
 
-	osImage, err = cachedImage(osImage)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to use cached libvirt image")
-	}
-
 	cfg := &config{
 		URI:          masterConfig.URI,
-		Image:        osImage,
 		IfName:       bridge,
 		BootstrapIP:  bootstrapIP.String(),
 		MasterIPs:    masterIPs,
 		MasterMemory: strconv.Itoa(masterConfig.DomainMemory),
 		MasterVcpu:   strconv.Itoa(masterConfig.DomainVcpu),
 	}
 
+	baseImageURL, err := url.ParseRequestURI(osImage)
+	osImage, err = cachedImage(osImage)
```

Comment:
As in the other comments, I wonder if we could put this decompression logic into either the libvirt cachedImage function, or the shared internal cache implementation?

Reply:
It's in the shared internal cache.